nixos/modules/services/web-apps/nifi.nix

```diff
@@ -0,0 +1,252 @@
+          test -f '/var/lib/nifi/conf/bootstrap.conf'                       || (cp '${cfg.package}/share/nifi/conf/bootstrap.conf' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/bootstrap.conf')
+          test -f '/var/lib/nifi/conf/bootstrap-hashicorp-vault.conf'       || (cp '${cfg.package}/share/nifi/conf/bootstrap-hashicorp-vault.conf' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/bootstrap-hashicorp-vault.conf')
+          test -f '/var/lib/nifi/conf/bootstrap-notification-services.xml'  || (cp '${cfg.package}/share/nifi/conf/bootstrap-notification-services.xml' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/bootstrap-notification-services.xml')
+          test -f '/var/lib/nifi/conf/logback.xml'                          || (cp '${cfg.package}/share/nifi/conf/logback.xml' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/logback.xml')
+          test -f '/var/lib/nifi/conf/login-identity-providers.xml'         || (cp '${cfg.package}/share/nifi/conf/login-identity-providers.xml' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/login-identity-providers.xml')
+          test -f '/var/lib/nifi/conf/nifi.properties'                      || (cp '${cfg.package}/share/nifi/conf/nifi.properties' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/nifi.properties')
+          test -f '/var/lib/nifi/conf/stateless-logback.xml'                || (cp '${cfg.package}/share/nifi/conf/stateless-logback.xml' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/stateless-logback.xml')
+          test -f '/var/lib/nifi/conf/stateless.properties'                 || (cp '${cfg.package}/share/nifi/conf/stateless.properties' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/stateless.properties')
+          test -f '/var/lib/nifi/conf/state-management.xml'                 || (cp '${cfg.package}/share/nifi/conf/state-management.xml' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/state-management.xml')
+          test -f '/var/lib/nifi/conf/zookeeper.properties'                 || (cp '${cfg.package}/share/nifi/conf/zookeeper.properties' '/var/lib/nifi/conf' && chmod 0640 '/var/lib/nifi/conf/zookeeper.properties')
+          ${optionalString ((cfg.initUser!=null) && (cfg.initPasswordFile!=null)) ''
+            awk -F'[<|>]' '/property name="Username"/ {if ($3!="") f=1} END{exit !f}' /var/lib/nifi/conf/login-identity-providers.xml || ${cfg.package}/bin/nifi.sh set-single-user-credentials ${cfg.initUser} $(cat ${cfg.initPasswordFile})
+          ''}
+          ${optionalString (cfg.enableHTTPS == false) ''
+            awk -F'=' '/^nifi.remote.input.secure=/ {if ($2=="false") f=1} END{exit !f}' /var/lib/nifi/conf/nifi.properties                   || sed -i -e 's|nifi.remote.input.secure=.*|nifi.remote.input.secure=false|g' /var/lib/nifi/conf/nifi.properties
```

Comment:
This looks super hacky. Is there no better way to write these properties with less code?

Reply:
This is a simple option that I managed to implement.
If generate a complete configuration file, then will not be able to manually change the parameters that are not specified in this module.
In this variant, we can change the main parameters, and the rest of the parameters can be changed manually.

Need a way that takes as a basis the default configuration file and changes certain parameters in it.

Reply:
There needs to be a better option than this. At least I think we should move this to an extra file.

Reply:
Allowed this variant?:
```
...
          ${optionalString ((cfg.initUser!=null) && (cfg.initPasswordFile!=null)) ''
            ${cfg.package}/bin/nifi.sh set-single-user-credentials ${cfg.initUser} $(cat ${cfg.initPasswordFile})
          ''}
          ${optionalString (cfg.enableHTTPS == false) ''
            sed -i -e 's|nifi.remote.input.secure=.*|nifi.remote.input.secure=false|g' /var/lib/nifi/conf/nifi.properties
            sed -i -e 's|nifi.web.http.host=.*|nifi.web.http.host=${cfg.listenHost}|g' /var/lib/nifi/conf/nifi.properties
            sed -i -e 's|nifi.web.http.port=.*|nifi.web.http.port=${(toString cfg.listenPort)}|g' /var/lib/nifi/conf/nifi.properties
...
```

Reply:
Updated `preStart` script.